### DISCUSSION TITLE
Default value for checked / unchecked in the `toggleRefinement` widget

### DISCUSSION THREAD

**Discussion by roshammar** [algolia/instantsearch#5065] [Ideas] [answered]
Hi,

I am using a `toggleRefinement` widget to allow users to filter on a specific attribute. However, its default state seems to be "off" / unchecked, but in my use case it makes more sense to have it "on" / checked by default. What would be the easiest way to accomplish this?

Would it be possible to add it as a new option?

I realise I could just invert my filter, but in my case it is not boolean, it is an attribute that can take several values. The toggle either selects a specific value or, when it is off, all possible values.

Cheers

**Comment by Haroenv** [comment, marked as answer]
```js
instantsearch({
  ...otherOptions,
  initialUiState: {
    myIndexName: {
      toggleRefinement: {
        myAttribute: true,
      },
    },
  },
});
```

you can use initialUiState or setUiState to set the default state of a widget

**Reply by roshammar** [reply]
Thank you very much, I had missed this possibility!
